api/src/main/java/marquez/db/DatasetDao.java

```diff
@@ -130,32 +147,51 @@ default void setFields(Dataset ds) {
   Optional<DatasetRow> getUuid(String namespaceName, String datasetName);
 
   @SqlQuery(
-      "SELECT d.*, dv.fields, sv.schema_location, facets, t.tags\n"
-          + "FROM datasets d\n"
-          + "INNER JOIN dataset_versions dv on d.current_version_uuid = dv.uuid\n"
-          + "LEFT OUTER JOIN stream_versions AS sv ON sv.dataset_version_uuid = dv.uuid\n"
-          + "LEFT OUTER JOIN (\n"
+      "WITH selected_datasets AS (\n"
```

Comment:
_Minor_: I _think_ it'd be possible to factor out some of the common SQL into a constant for reuse?

Reply:
yeah, we need to start incorporating some strategy to reuse SQL- I felt this badly when I was implementing the jobVersion/datasetVersions in the Run model.

Here, the `selected_datasets` differs between the two queries- one specifies namespace and name, the other just specifies namespace and applies limit/offset. The rest of the SQL is the same until the end- the `findAll` query applies an `ORDER BY` clause at the end. It seems silly to me to extract the middle out and concatenate, so my preference is to leave it as is until we either start organizing reusable sql or refactoring tables so we don't need quite so much.

Reply:
Makes sense! I'll open an issue to begin evaluating some of the current SQL and a plan to support reusability. But, for now, this is great 👍